eZ/Publish/Core/FieldType/BinaryBase/BinaryBaseStorage.php

```diff
@@ -107,7 +107,9 @@ public function copyLegacyField(VersionInfo $versionInfo, Field $field, Field $o
         }
 
         // field translations have their own file reference, but to the original file
-        $originalField->value->externalData['id'];
+        $field->value->externalData['id'] = $originalField->value->externalData['id'];
```

Comment:
Is there anything else in externalData that for some reason we dont want to copy?

Reply:
@ViniTou  the problem with `getFieldData` is that it is used to 'read' the file at the first time (so when we are actually storing the file), later on, when the field is copied it is not fired at all as the data is copied from an existing field.

Reply:
I just wonder if we can have `$field->value->externalData = $originalField->value->externalData` and be sure that always all properties are copied. But if one if left out not sure if we should then.

Reply:
I think we shouldn't just for the sake of `downloadCount`.